### DISCUSSION TITLE
Would you explain in technical details why "Self-extracting ZIP" can't be opened without SingleFile while "Self-extracting ZIP (universal)" can?

### DISCUSSION THREAD

**Discussion by StoopidoMan** [gildas-lormeau/SingleFile#1269] [Q&A]
Would you explain in technical details why "Self-extracting ZIP" can't be opened without SingleFile while "Self-extracting ZIP (universal)" can?

The reason I ask because it will help me in my decision of choosing between both.

Thank you for your amazing Addon <3 StoopidoMan thank you

**Comment by gildas-lormeau** [comment]
From a technical point of view, the existence of the "universal' format is due to the fact that Chromium-based and WebKit-based browers don't allow a script to execute `fetch("")` in a page opened from the filesystem, for "security reasons". The universal method allows the script to circumvent this issue by reading the ZIP binary data directly from the DOM.

**Comment by cloudrac3r** [comment]
This is such a cool feature to add. Thank you for creating and continuing to update this great browser extension :)

**Comment by StoopidoMan** [comment]
> From a technical point of view, the existence of the "universal' format is due to the fact that Chromium-based and WebKit-based browers don't allow a script to execute `fetch("")` in a page opened from the filesystem, for "security reasons". The universal method allows the script to circumvent this issue by reading the ZIP binary data directly from the DOM.

would you confirm if i am right after extracting the content of "Self-extracting ZIP" and "Self-extracting ZIP (universal)" they are both identical. is that correct?

**Suggestion:**
prefixing the format with extensions:

`u.zip.html`

and 

`zip.html`

because in far future (10 years) the user will remember that it is a se;f extracting zip file with an html extension and also when shared with others the ".zip" prefix will help them know that it might be a zip file. 


**don't read after this line @gildas-lormeau, it is for other users of SingleFile:**

The reason I asked that question, is for archivability and convenience. 
I personally think for long term storage "Self-Extracting Zip" is a better format because as long as both format are exactly the same then in future if SingleFile ceased to exist, you will just extract the content and you will gain 1% more storage. 

And if you might share a file then choose the universal format. 


Again @gildas-lormeau THANK YOU VERY MUCH FOR THIS AMAZING TOOL

**Comment by gildas-lormeau** [comment]
> would you confirm if i am right after extracting the content of "Self-extracting ZIP" and "Self-extracting ZIP (universal)" they are both identical. is that correct?

I confirm you're right. The universal version simply includes some additional data (outside the zip file) to consolidate the missing data read in the DOM.

> **Suggestion:** prefixing the format with extensions:
> 
> `u.zip.html`
> 
> and
> 
> `zip.html`
> 

The problem is that it means the template would be automatically modified. I think the solution might be to introduce a new template variable returning the extension depending on the options.

> Again @gildas-lormeau THANK YOU VERY MUCH FOR THIS AMAZING TOOL

Thank you for the kind words and the suggestion as well :)

**Comment by gildas-lormeau** [comment]
The file name extension issue is fixed in the latest version via the new template variable `{filename-extension}`.
